### PR TITLE
Add apply license choices for AsciiDocTemplateReporter

### PR DESCRIPTION
**Pull request by MarcelBochtler**
Warning: This changes the output of the disclosure document as license choices are now applied, without actually changing the template.

### CODE REVIEW

File: evaluator/src/main/kotlin/PackageRule.kt [hunk 1]
```diff
@@ -136,7 +136,7 @@ open class PackageRule(
      */
```
Comment (oheger-bosch) [inline evaluator/src/main/kotlin/PackageRule.kt:136]: Commit message: The body text is a bit strange, there seems to be a preposition too much. Maybe change to something like "Rename the method to make a clear distinction between repository-wide and package-local license choices." ?

File: reporter/src/main/kotlin/utils/FreemarkerTemplateProcessor.kt [hunk 2]
```diff
@@ -251,20 +262,27 @@ class FreemarkerTemplateProcessor(
                 .get(LicenseView.Companion) as LicenseView
 
         /**
-         * Merge the [ResolvedLicense]s of multiple [models] and filter them using [licenseView]. [Omits][omitExcluded]
-         * excluded packages, licenses, and copyrights by default. [Undefined][omitNotPresent] licenses can be filtered
-         * out optionally. The returned list is sorted by license identifier.
+         * Merge the [ResolvedLicense]s of multiple [models] and filter them using [licenseView] and
+         * [PackageModel.licenseChoices]. [Omits][omitExcluded] excluded packages, licenses, and copyrights by default.
+         * [Undefined][omitNotPresent] licenses can be filtered out optionally. [LicenseChoices][skipLicenseChoices] are
+         * applied by default The returned list is sorted by license identifier.
```
Comment (oheger-bosch) [inline reporter/src/main/kotlin/utils/FreemarkerTemplateProcessor.kt:274]: Missing '.' after "by default".

**Review by oheger-bosch — COMMENTED** [review]
Changes look good to me.

The only thing I am not sure about is whether this functionality should be implemented in the template processor. Would it make sense to have it available more centrally? Having a view on licenses with license choices applied could be useful for other reports or maybe rules as well.

**Review by oheger-bosch — DISMISSED** [review]

File: reporter/src/main/kotlin/utils/FreemarkerTemplateProcessor.kt [hunk 3]
```diff
@@ -218,9 +218,12 @@ class FreemarkerTemplateProcessor(
          * file and all licenses not contained in those files shall be listed separately.
          */
         @Suppress("UNUSED") // This function is used in the templates.
-        fun licensesNotInLicenseFiles(): List<ResolvedLicense> {
+        @JvmOverloads
```
Comment (sschuberth) [inline reporter/src/main/kotlin/utils/FreemarkerTemplateProcessor.kt:221]: Commit message nit: s/asciidoc/AsciiDoc/

File: model/src/main/kotlin/licenses/LicenseView.kt [hunk 4]
```diff
@@ -22,6 +22,7 @@ package org.ossreviewtoolkit.model.licenses
 import org.ossreviewtoolkit.model.LicenseSource
 import org.ossreviewtoolkit.model.Package
 import org.ossreviewtoolkit.spdx.SpdxSingleLicenseExpression
+import org.ossreviewtoolkit.spdx.model.LicenseChoice
```
Comment (sschuberth) [inline model/src/main/kotlin/licenses/LicenseView.kt:25]: Commit message: I had to read the body several times to understand it. For me, a comma after "LicenseView" would have helped.

**Review by sschuberth — APPROVED** [review]

**Review by sschuberth — APPROVED** [review]

**Pull request merged**